spinn_front_end_common/interface/interface_functions/spalloc_allocator.py

```diff
@@ -57,7 +57,8 @@ def _wait(self):
 
     @overrides(MachineAllocationController._teardown)
     def _teardown(self):
-        self._job.close()
+        if not self._exited:
```

Comment:
Needs to call superclass implementation of this method. _After_ this test/`if`.

Reply:
NB: the method was wrong before. Doesn't mean it should stay wrong.

Reply:
I guess it should - teardown is a "pass" method just now, but it isn't abstract so it doesn't have to stay that way...

Reply:
Done - this happens regardless of whether the if is met (which I think is correct).

Reply:
It only mattered because that super call is how the `_exited` field gets set.